component/s3storage/client.go

```diff
@@ -138,15 +106,18 @@ func (cl *Client) Configure(cfg Config) error {
 	if cl.Config.usePathStyle {
 		cl.awsS3Client = s3.NewFromConfig(defaultConfig, func(o *s3.Options) {
 			o.UsePathStyle = true
+			o.BaseEndpoint = aws.String(cl.Config.authConfig.Endpoint)
 		})
 	} else {
-		cl.awsS3Client = s3.NewFromConfig(defaultConfig)
+		cl.awsS3Client = s3.NewFromConfig(defaultConfig, func(o *s3.Options) {
+			o.BaseEndpoint = aws.String(cl.Config.authConfig.Endpoint)
+		})
 	}
 
-	// ListBuckets here to test connection
-	_, err = cl.ListBuckets()
+	// List objects to test connection
+	_, _, err = cl.List("/", nil, 1)
 	if err != nil {
-		log.Err("Client::Configure : listing buckets failed. Here's why: %v", err)
+		log.Err("Client::Configure : listing objects failed. Here's why: %v", err)
```

Comment:
I'm a bit confused about this... Where is the bucket set?
And is there a reason this needs to change?
If we default to the first visible bucket, we should probably only do that when the bucket-name config option is empty.

Reply:
The bucket is set whenever we make an S3 call and it is set in the config. With the new default to the first bucket, this additional list call validates that the region is correct (which ListBuckets does not do) and validates that we have access to the bucket.